codelab/JinghangCodeLab/MagicNumberConverter.java

```diff
@@ -0,0 +1,55 @@
+package JinghangCodeLab;
+
+public class MagicNumberConverter {
+
+    public static int binaryToDecimal(String binary) {
+        if (Double.parseDouble(binary) > Integer.MAX_VALUE){
+            System.out.println("Out of Range not in Integer Range");
```

Comment:
you can thrown an exception in this case.

throw new OutOfRangeException("Input binary number is of out of range, input binary number=" + binary);